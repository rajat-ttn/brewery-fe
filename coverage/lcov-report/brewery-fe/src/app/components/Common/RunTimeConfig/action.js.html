<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for brewery-fe/src/app/components/Common/RunTimeConfig/action.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../../../index.html">All files</a> / <a href="index.html">brewery-fe/src/app/components/Common/RunTimeConfig</a> action.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>2/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>2/4</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import ACTION from '../../../../constants/actions';
&nbsp;
export const setTemperatureType = <span class="fstat-no" title="function not covered" >(c</span>urrentValue) =&gt; {
<span class="cstat-no" title="statement not covered" >    return {</span>
        type: ACTION.SET_TEMPERATURE_TYPE,
        payload: currentValue,
    }
};
&nbsp;
export const toggleSound = <span class="fstat-no" title="function not covered" >(i</span>sMute) =&gt; {
<span class="cstat-no" title="statement not covered" >    return {</span>
        type: ACTION.TOGGLE_SOUND,
        payload: isMute,
    }
};
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat Mar 10 2018 17:04:54 GMT+0530 (IST)
</div>
</div>
<script src="../../../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../../../sorter.js"></script>
</body>
</html>
