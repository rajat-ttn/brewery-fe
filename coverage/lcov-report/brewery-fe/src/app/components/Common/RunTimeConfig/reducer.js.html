<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for brewery-fe/src/app/components/Common/RunTimeConfig/reducer.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../../../index.html">All files</a> / <a href="index.html">brewery-fe/src/app/components/Common/RunTimeConfig</a> reducer.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">57.14% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>4/7</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">57.14% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/7</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import ACTION from '../../../../constants/actions';
const initialState = {
    temperatureType : 'CELSIUS',
    isMute: false,
};
&nbsp;
const setTemperatureType = <span class="fstat-no" title="function not covered" >(s</span>tate, payload) =&gt; {
<span class="cstat-no" title="statement not covered" >    return { ...state, temperatureType:payload };</span>
};
&nbsp;
export default function configReducer(state = initialState, action) {
    switch (action.type) {
<span class="branch-0 cbranch-no" title="branch not covered" >        case ACTION.SET_TEMPERATURE_TYPE: <span class="cstat-no" title="statement not covered" >return setTemperatureType(state, action.payload);</span></span>
<span class="branch-1 cbranch-no" title="branch not covered" >        case ACTION.TOGGLE_SOUND: <span class="cstat-no" title="statement not covered" >return { ...state, isMute: action.payload };</span></span>
        default: return state;
    }
};
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat Mar 10 2018 17:04:54 GMT+0530 (IST)
</div>
</div>
<script src="../../../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../../../sorter.js"></script>
</body>
</html>
