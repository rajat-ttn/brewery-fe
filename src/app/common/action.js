import ACTION from '../../constants/actions';

export const updateBeerTemperature = (data) => {
    return {
        type: ACTION.UPDATE_BEER_TEMPERATURE,
        payload: data,
    }
};