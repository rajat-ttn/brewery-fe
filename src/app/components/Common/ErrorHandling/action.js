import ACTION from '../../../../constants/actions';

// function to update error message in redux state
export const setErrorMessage = (error) => {
    return {
        type: ACTION.SET_ERROR,
        payload: error,
    }
};

// function to remove error message from redux state
export const removeErrorMessage = () => {
    return {
        type: ACTION.REMOVE_ERROR
    }
};
