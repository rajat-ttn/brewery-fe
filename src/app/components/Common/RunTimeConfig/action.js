import ACTION from '../../../../constants/actions';

// function to set Temperature type in redux state
export const setTemperatureType = (currentValue) => {
    return {
        type: ACTION.SET_TEMPERATURE_TYPE,
        payload: currentValue,
    }
};

// function to set playing status of sound in redux state
export const toggleSound = (isMute) => {
    return {
        type: ACTION.TOGGLE_SOUND,
        payload: isMute,
    }
};
