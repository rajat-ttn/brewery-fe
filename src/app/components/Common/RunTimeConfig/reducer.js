import ACTION from '../../../../constants/actions';

const initialState = {
    temperatureType: 'CELSIUS',
    isMute: false,
};

const setTemperatureType = (state, payload) => {
    return { ...state, temperatureType: payload };
};

export default function configReducer(state = initialState, action) {
    switch (action.type) {
        case ACTION.SET_TEMPERATURE_TYPE: return setTemperatureType(state, action.payload);
        case ACTION.TOGGLE_SOUND: return { ...state, isMute: action.payload };
        default: return state;
    }
};
