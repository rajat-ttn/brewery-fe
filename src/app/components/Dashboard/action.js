import ACTION from '../../../constants/actions';

// Action Creater to Update list of containers
export const getBeerList = (beerList) => {
    return {
        type: ACTION.GET_BEER_LIST,
        payload: beerList,
    }
};

// Action Creater to Update Current Temperature of container
export const updateBeerTemperature = (data) => {
    return {
        type: ACTION.UPDATE_CONTAINER_TEMPERATURE,
        payload: data,
    }
};
