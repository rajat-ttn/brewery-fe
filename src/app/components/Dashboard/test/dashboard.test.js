import React from 'react';
import { fetchBeerList } from '../async.action';
import { setTemperatureType } from '../../Common/RunTimeConfig/action';
import { shallow } from 'enzyme';
import {DashboardContainer, mapStateToProps, mapDispatchToProps} from '../index';


jest.mock('../async.action');
const FETCH_ACTION = Symbol('');
fetchBeerList.mockReturnValue(FETCH_ACTION);


describe('BeerComponent component renders the BeerComponent correctly', () => {
    let allProps;

    beforeEach(() => {
        allProps = {
            beerList: [{beerType : "Aler Beer", containerId: 1, id: 1, tempRange: [3,5]}],
            temperatureType: "FAHRENHEIT",
            fetchBeerList: () => {},
            setTemperatureType: () => {}
        };
    });


    it('renders correctly with beer detail and temperature', () => {
        expect(<DashboardContainer {...allProps}  />).toMatchSnapshot();
    });

    it('renders correctly without beerList', () => {
        allProps.beerList = [];
        expect(<DashboardContainer {...allProps}  />).toMatchSnapshot();
    });

    it('renders correctly with different temperature', () => {
        allProps.temperatureType = "CELSIUS";
        expect(<DashboardContainer {...allProps}  />).toMatchSnapshot();
    });
});

describe('mapStateToProps ', () => {
    const state = {
        beerList: [{ a: 20 }],
        temperatureType: {
            temperatureType: 'CELSIUS',
        }
    };

    it('should render correct object', () => {
        expect(mapStateToProps(state)).toEqual({
            beerList: state.beerList,
            temperatureType: state.temperatureType.temperatureType,
        });
    });
});

describe('mapDispatchToProps', () => {
    const dispatch = jest.fn().mockImplementation(x => x);
    const actionDispatchers = mapDispatchToProps(dispatch);

    it('should render the correct list of methods', () => {
       expect(mapDispatchToProps()).toEqual({
           fetchBeerList: expect.any(Function),
           setTemperatureType: expect.any(Function)
       })
    });

    it('returns the result of dispatching fetchBeerList correctly', () => {
        expect(actionDispatchers.fetchBeerList()).toBe(FETCH_ACTION);
        expect(dispatch).toHaveBeenCalledWith(FETCH_ACTION);
    });
});





