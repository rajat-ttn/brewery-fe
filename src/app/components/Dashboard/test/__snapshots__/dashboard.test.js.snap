// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeerComponent component renders the BeerComponent correctly renders correctly with beer detail and temperature 1`] = `
<Dashboard
  beerList={
    Array [
      Object {
        "beerType": "Aler Beer",
        "containerId": 1,
        "id": 1,
        "tempRange": Array [
          3,
          5,
        ],
      },
    ]
  }
  fetchBeerList={[Function]}
  setTemperatureType={[Function]}
  temperatureType="FAHRENHEIT"
/>
`;

exports[`BeerComponent component renders the BeerComponent correctly renders correctly with different temperature 1`] = `
<Dashboard
  beerList={
    Array [
      Object {
        "beerType": "Aler Beer",
        "containerId": 1,
        "id": 1,
        "tempRange": Array [
          3,
          5,
        ],
      },
    ]
  }
  fetchBeerList={[Function]}
  setTemperatureType={[Function]}
  temperatureType="CELSIUS"
/>
`;

exports[`BeerComponent component renders the BeerComponent correctly renders correctly without beerList 1`] = `
<Dashboard
  beerList={Array []}
  fetchBeerList={[Function]}
  setTemperatureType={[Function]}
  temperatureType="FAHRENHEIT"
/>
`;
